presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -299,42 +298,31 @@ public Expression getJoinCriteria(Join join)
 
     public void recordSubqueries(Node node, ExpressionAnalysis expressionAnalysis)
     {
-        this.inPredicatesSubqueries.putAll(node, expressionAnalysis.getSubqueryInPredicates());
-        this.scalarSubqueries.putAll(node, expressionAnalysis.getScalarSubqueries());
-        this.existsSubqueries.putAll(node, expressionAnalysis.getExistsSubqueries());
-        this.quantifiedComparisonSubqueries.putAll(node, expressionAnalysis.getQuantifiedComparisons());
+        NodeRef<Node> key = NodeRef.of(node);
+        this.inPredicatesSubqueries.putAll(key, expressionAnalysis.getSubqueryInPredicates());
+        this.scalarSubqueries.putAll(key, expressionAnalysis.getScalarSubqueries());
+        this.existsSubqueries.putAll(key, expressionAnalysis.getExistsSubqueries());
+        this.quantifiedComparisonSubqueries.putAll(key, expressionAnalysis.getQuantifiedComparisons());
     }
 
     public List<InPredicate> getInPredicateSubqueries(Node node)
     {
-        if (inPredicatesSubqueries.containsKey(node)) {
-            return inPredicatesSubqueries.get(node);
-        }
-        return ImmutableList.of();
+        return ImmutableList.copyOf(inPredicatesSubqueries.get(NodeRef.of(node)));
     }
 
     public List<SubqueryExpression> getScalarSubqueries(Node node)
     {
-        if (scalarSubqueries.containsKey(node)) {
-            return scalarSubqueries.get(node);
-        }
-        return ImmutableList.of();
+        return ImmutableList.copyOf(scalarSubqueries.get(NodeRef.of(node)));
     }
 
     public List<ExistsPredicate> getExistsSubqueries(Node node)
     {
-        if (existsSubqueries.containsKey(node)) {
-            return existsSubqueries.get(node);
-        }
-        return ImmutableList.of();
+        return ImmutableList.copyOf(existsSubqueries.get(NodeRef.of(node)));
     }
 
     public List<QuantifiedComparisonExpression> getQuantifiedComparisonSubqueries(Node node)
     {
-        if (quantifiedComparisonSubqueries.containsKey(node)) {
-            return quantifiedComparisonSubqueries.get(node);
-        }
-        return ImmutableList.of();
+        return ImmutableList.copyOf(quantifiedComparisonSubqueries.get(NodeRef.of(node)));
```

Comment:
`unmodifiableView`? No need to copy the list. It's used during planning but build during analysis.